akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -127,9 +130,10 @@ private[http] final class BodyPartParser(
         // we don't use boyerMoore here because we are testing for the boundary *without* a
         // preceding CRLF and at a known location (the very beginning of the entity)
         try {
-          if (boundary(input, 0)) {
-            val ix = boundaryLength
-            if (crlf(input, ix)) parseHeaderLines(input, ix + 2)
+          eolConfiguration = EndOfLineConfiguration(input, boundary)
```

Comment:
I guess it's a valid approach to guess the EOL config from the first boundary. We still won't support mixed EOL style but I guess that's ok.